### PR TITLE
hexer 1.4.0 (new formula)

### PR DESCRIPTION
**Pull request by mabrowning**
Hexer is a LGPL C++ library that provides some classes for generating hexbin density surfaces and multipolygon boundaries for large point sets.

Simple cmake project with 1 dependency (gdal)

I'm starting to add all the dependencies needed to build [OpenDroneMap](opendronemap.org) natively. This library is needed to build a plugin for the `pdal` package (pull request forthcoming).

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by mabrowning** [comment]
Build failed because github repo doesn't have 50 stars... currently at 42. However, this is a dependency for [OpenDroneMap](https://github.com/OpenDroneMap/OpenDroneMap/), which has hundreds of stars, forks, and follows.

**Comment by fxcoudert** [comment]
@mabrowning Do you have plans to submit OpenDroneMap as a formula shortly?

**Comment by mabrowning** [comment]
@fxcoudert I am, although there are quite a few packages I'll need to formulated, and I'll need to work out how best to get all the python dependencies available from pip installed.

For reference, I'll be submitting or extending:
pdal (to include this hexer plugin)
[OpenGV](https://github.com/paulinus/opengv/)
[OpenSfM](https://github.com/mapillary/OpenSfM/)
[ecto](https://github.com/plasmodic/ecto) a python work queue framework - although this may be better distributed through pypi
[OpenDroneMap](https://github.com/OpenDroneMap/OpenDroneMap)

in roughly that dependency order. Alternatively, I could package all of these up as private dependencies in the OpenDroneMap formula, but I think OpenSfM (which depends on OpenGV) does have wider appeal.

**Comment by mabrowning** [comment]
I should clarify. OpenDroneMap calls a series of executables to perform much of its work. One of those is `pdal`, which homebrew has a package for. However, `pdal` is setup as a series of (optional) plugins, and OpenDroneMap makes use of a plugin which the homebrew package doesn't include. This `hexer` package is necessary so that we can modify the `pdal` package to include the `hexbin` plugin for pdal (see [pdal-add-hexbin](https://github.com/mabrowning/homebrew-core/tree/pdal-add-hexbin) which I have not submitted yet as it depends on this package).

**Comment by fxcoudert** [comment]
Some of these seem quite low usage in themselves… and some don't even have released versions, which is going to be a problem for Homebrew. Another option, as you said, would be to vendor them as private dependencies inside your final formula.

**Comment by mabrowning** [comment]
I agree, especially after looking at a few other small-time packages that OpenDroneMap makes use of. The reason I'm starting with hexer is that is requires modification of an existing homebrew package. All others can be achieved as vender dependencies inside of OpenDroneMap. 

However, I do think that OpenSfM would be generally useful to many more users than even OpenDroneMap. That package is widely used and versioned.

File: Formula/hexer.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class Hexer < Formula
+  desc "Hexbin density and boundary generation"
+  homepage "https://github.com/hobu/hexer"
+  url "https://github.com/hobu/hexer/archive/1.4.0.tar.gz"
+  sha256 "886134fcdd75da2c50aa48624de19f5ae09231d5290812ec05f09f50319242cb"
+  depends_on "cmake" => :build
```
Comment (fxcoudert) [inline Formula/hexer.rb]: Empty line before that

File: Formula/hexer.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  depends_on "gdal"
+  def install
```
Comment (fxcoudert) [inline Formula/hexer.rb]: Empty line before that

File: Formula/hexer.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    system "cmake", ".", *std_cmake_args
+    system "make", "install"
+  end
+  test do
```
Comment (fxcoudert) [inline Formula/hexer.rb]: Empty line before that

File: Formula/hexer.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    system bin/"curse", "-c", "hex", "-i", "fake_file"
```
Comment (fxcoudert) [inline Formula/hexer.rb]: Can we check the output? Or perform some operation on a real file, and check that it works?

**Review by fxcoudert — CHANGES_REQUESTED** [review]

**Comment by fxcoudert** [comment]
Please squash all changes to be made in a single commit, with message `hexer 1.4.0 (new formula)`

**Pull request closed without merging**